web/src/main/java/org/springframework/security/web/authentication/rememberme/TokenBasedRememberMeServices.java

```diff
@@ -172,15 +217,33 @@ public void onLoginSuccess(HttpServletRequest request, HttpServletResponse respo
 		long expiryTime = System.currentTimeMillis();
 		// SEC-949
 		expiryTime += 1000L * ((tokenLifetime < 0) ? TWO_WEEKS_S : tokenLifetime);
-		String signatureValue = makeTokenSignature(expiryTime, username, password);
-		setCookie(new String[] { username, Long.toString(expiryTime), signatureValue }, tokenLifetime, request,
-				response);
+		String signatureValue = makeTokenSignature(expiryTime, username, password, this.encodingAlgorithm);
+		setCookie(new String[] { username, Long.toString(expiryTime), this.encodingAlgorithm.name(), signatureValue },
+				tokenLifetime, request, response);
 		if (this.logger.isDebugEnabled()) {
 			this.logger.debug(
 					"Added remember-me cookie for user '" + username + "', expiry: '" + new Date(expiryTime) + "'");
 		}
 	}
 
+	/**
+	 * Sets the algorithm to be used to encode the token signature
+	 * @param encodingAlgorithm the encoding algorithm
```

Comment:
I think that https://github.com/spring-projects/spring-security/pull/11464#pullrequestreview-1031035009 makes a good point about adding this to the constructor. I see now also that this is how `DelegatingPasswordEncoder` is designed, with the encoding encoder being set in the constructor and the matching encoder being set as a setter.